Fail with ENOSYS on unsupported socketcalls

ngg opened the pull request:
There are some socketcalls that musl can call and are not handled
(for example socketpair). These currently abort the program instead of
returning an error code.

sbc100 commented on src/library_syscall.js:
```diff
@@ -720,7 +720,12 @@ var SyscallsLibrary = {
-      default: abort('unsupported socketcall syscall ' + call);
+      default: {
+#if SYSCALL_DEBUG
+        err('    (socketcall: ' + call + ')');
```
why drop the "unsupported" part?

ngg replied:
In this same function the other err() functions used this style. I can revert back the format if you'd like.

ngg commented:
> I think this makes sense. Can you add/update test cases?
I've added a test case for socketpair() and also fixed function dependencies to make that test case (and other similarly simple programs) compile.
